src/lib/telemetry.rs

```diff
@@ -228,7 +302,35 @@ macro_rules! info_and_incr {
     }
 }
 
-/// TODO doc
+/// Create an error-level log trace and increment a statsd counter with the same
+/// name.
+///
+/// The macro expects a `StatsD` client as its first argument, followed by a
+/// `LogKey` enum value. Aside from this, the macro behaves exactly like the
+/// `error!` macro.
+///
+/// # Examples
+///
+/// ```
+/// use std::error::Error;
+///
+/// let statsd = StatsD::new(&settings);
+///
+/// error_and_incr!(
+///     statsd,
+///     LogKey::CleanupAicArchiveFailed,
+///     key = "value",
+///     "Some log message"
+/// )
```

Comment:
add a note about the format of the statsd key that gets created e.g. cjms.cleanup-aic-archive-failed